src/main/java/command/DoneCommand.java

```diff
@@ -0,0 +1,58 @@
+package command;
+import task.*;
+import duke.*;
+
+/**
+ * DoneCommand represents a done command.
+ *
+ * @author Ho Wen Zhong
+ */
+public class DoneCommand extends Command {
+
+    private int doneIndex;
+
+    public DoneCommand(int doneIndex) {
+        this.doneIndex = doneIndex;
+    }
+
+    /**
+     * Sets task as done in TaskList, calls showResponse with response,
+     * saves TaskList to file.
+     *
+     * @param tasks Current TaskList.
+     * @param ui Ui object.
+     * @param storage Storage object.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        tasks.done(doneIndex);
+
+        Task task = tasks.get(doneIndex);
```

Comment:
If the index is negative or too big, it will throw an IndexOutOfBoundsException. Should probably check for it.